Runner/AssemblyScanner.cs

```diff
@@ -0,0 +1,112 @@
+﻿using Gauge.CSharp.Lib;
+using Gauge.CSharp.Lib.Attribute;
+using NLog;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace Gauge.CSharp.Runner
+{
+    public class AssemblyScanner
+    {
+        private static readonly Logger Logger = LogManager.GetLogger("AssemblyScanner");
+        private Dictionary<Type, List<MethodInfo>> AttributeToMethodInfo;
```

Comment:
I'd suggest the following:

* Create a constructor taking a set of attribute types to scan for.
* Rename the `TryAdd` function to `ScanAndLoad`
* Return dictionary from attribute to List<MethodInfo> in `ScanAndLoad`

So `AssemblyScanner` only knows about the attributes to scan for and does not need to hold the assemblies and the Dictionary.